app/services/platform/submission.rb

```diff
@@ -0,0 +1,37 @@
+module Platform
+  class Submission
+    include ActiveModel::Model
+    attr_accessor :service, :user_data
+
+    REQUIRED_ENV_VARS = %w(
+      SERVICE_EMAIL_PDF_HEADING
+      SERVICE_EMAIL_PDF_SUBHEADING
+      SERVICE_EMAIL_OUTPUT
+      SERVICE_EMAIL_SENDER
+      SERVICE_EMAIL_SUBJECT
+      SERVICE_EMAIL_BODY
+    ).freeze
+
+    validate do
+      if REQUIRED_ENV_VARS.any? { |env_var| ENV[env_var].blank? }
+        Rails.logger.info(
+          'Ignoring submission. There are env vars missing: %{env_vars}' % {
+            env_vars: REQUIRED_ENV_VARS.select { |env_var| ENV[env_var].blank? }
+          }
+        )
+        self.errors.add(:base)
+      end
```

Comment:
I added into the commit message. I added this but I am not sure if we should. There are many things that can go wrong here. 

Basically the question is: should we validate as I am validating here?

The problems with the code above is:

1. If we only send the submission "if the env vars are there"
2. Then there is the possibility of any bug on the editor to cause losses of data because users will submit and nothing will happen -> if the required env vars are not there

Just marking this commit so we can pair and re-visit this topic.

Reply:
So I changed to only require the `SERVICE_EMAIL_OUTPUT` then we are behaving in the same way as the classic runner ™️